src/main/java/yale/Storage.java

```diff
@@ -0,0 +1,52 @@
+package yale;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private final File file;
+
+    /**
+     * Creates a Storage which can read and write to a task file.
+     *
+     * @param filename Filename of the task file.
+     */
+    public Storage(String filename) {
+        this.file = new File(filename);
+    }
+
+    /**
+     * Reads in the tasks in the task file and outputs them.
+     *
+     * @return ArrayList of the tasks in the file.
+     */
+    public ArrayList<Task> readTasks() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try (Scanner in = new Scanner(file)) {
+            while (in.hasNextLine()) {
+                Task task = Task.fromCsv(in.nextLine());
+                if (task != null) {
+                    tasks.add(task);
+                }
+            }
+        } catch (Exception ignored) {}
```

Comment:
```suggestion
        } catch (Exception ignored) {
            throw new YaleException(ignored.getMessage);
        }
```

i think it would be good to throw an exception to surface up to the Yale bot so that the user can see that there is a problem reading the file if any

but between you and me, yolo